# Individual homework project 2

You will apply statistical testing and correlation analysis to the ShopNow dataset.

For each question:

1. Run the appropriate test.
2. Report the test statistic and p-value (if applicable).
3. Interpret the result in plain English.
4. Explain what it means for the business.

### Import modules

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import ttest_ind, chi2_contingency, pearsonr

### Import datasets

In [2]:
customers = pd.read_csv("C:\\Users\\User pc\\Desktop\\MayerfeldPracticum\\data\\customers.csv")
customers.head()

,customer_id,age,gender,country,acquisition_channel,signup_date
0,1,56.0,Other,Germany,referral,2019-07-21
1,2,38.0,Other,Netherlands,email,2022-03-04
2,3,28.0,Other,Australia,referral,2024-05-21
3,4,19.0,Female,Australia,paid_search,2022-06-28
4,5,38.0,Female,Australia,referral,2024-07-17


In [3]:
transactions = pd.read_csv("C:\\Users\\User pc\\Desktop\\MayerfeldPracticum\\data\\transactions.csv")
transactions.head()

,transaction_id,customer_id,product_category,purchase_value,discount_used,purchase_date
0,1,8141,home,74.32,0,2023-09-05
1,2,1297,fashion,17.51,0,2019-08-07
2,3,7930,toys,39.96,1,2020-02-07
3,4,2426,electronics,43.79,0,2020-10-02
4,5,6614,fashion,187.01,0,2023-09-15


In [4]:
web_sessions = pd.read_csv("C:\\Users\\User pc\\Desktop\\MayerfeldPracticum\\data\\web_sessions.csv")
web_sessions.head()

,session_id,customer_id,pages_viewed,time_on_site,device_type,traffic_source,session_date,converted
0,1,9716,8,271.238207,desktop,organic,2022-06-18,0
1,2,4617,5,257.386842,mobile,social,2020-07-03,0
2,3,1506,4,289.679888,tablet,paid,2020-12-12,0
3,4,3869,5,146.543376,mobile,direct,2023-09-12,0
4,5,6245,3,339.228209,mobile,email,2019-04-02,0


In [5]:
customers_transactions = transactions.merge(customers, on='customer_id')
customers_transactions.head()

,transaction_id,customer_id,product_category,purchase_value,discount_used,purchase_date,age,gender,country,acquisition_channel,signup_date
0,1,8141,home,74.32,0,2023-09-05,39.0,Male,UK,social,2021-04-13
1,2,1297,fashion,17.51,0,2019-08-07,43.0,Female,Spain,email,2024-03-10
2,3,7930,toys,39.96,1,2020-02-07,19.0,Male,Netherlands,referral,2022-11-28
3,4,2426,electronics,43.79,0,2020-10-02,67.0,Female,Australia,referral,2023-09-14
4,5,6614,fashion,187.01,0,2023-09-15,50.0,Other,Germany,paid_search,2021-10-20


### Specific questions

### Q1. Do customers from the UK spend more per transaction than customers from Germany?

In [6]:
Germany = customers_transactions[customers_transactions["country"] == 'Germany']["purchase_value"]
UK = customers_transactions[customers_transactions["country"] == 'UK']["purchase_value"]

t_stat, p_value = ttest_ind(Germany, UK)

print(f"t-statistic: {t_stat:.3f}")
print(f"p-value: {p_value:.5f}")


t-statistic: 0.112
p-value: 0.91082


    H0 : Their is no significant difference in mean purchase value of Germany and UK
    
    H1 : Their is a significant difference in mean purchase value of Germany and UK
    
    Since p-value of 0.91082 is greater than 0.05(alpha value)
    
    Conclusion: we accept the null Hypothesis(H0) that their is no significant difference in mean purchase value of Germany     and UK
    
    Meaning that: customers from UK are not spending more than customers from Germany

### Q2. Do discounted purchases differ in average value from non-discounted purchases?

In [7]:
discount = transactions[transactions["discount_used"] == 1]["purchase_value"]
no_discount = transactions[transactions["discount_used"] == 0]["purchase_value"]

t_stat, p_value = ttest_ind(discount, no_discount)

print(f"t-statistic: {t_stat:.3f}")
print(f"p-value: {p_value:.5f}")


t-statistic: -27.160
p-value: 0.00000


    H0 : Their is no significant difference in mean purchase value of the discounted and non-discounted

    H1 : Their is a significant difference in mean purchase value of the discounted and non-discounted

    Since p-value of 0.00000 is less than  0.05(alpha value)

    Conclusion: we accept the alternative Hypothesis(H1) that their is a significant difference in mean purchase value of 
    the discounted and non-discounted
    
    Meaning: the discounted average purchase value differs to the average purchase value of non-discounted

### Q3. Do male and female customers differ in average purchase value?

In [8]:
male = customers_transactions[customers_transactions["gender"] == 'Male']["purchase_value"]
female = customers_transactions[customers_transactions["gender"] == 'Female']["purchase_value"]

t_stat, p_value = ttest_ind(male, female)

print(f"t-statistic: {t_stat:.3f}")
print(f"p-value: {p_value:.5f}")



t-statistic: 1.083
p-value: 0.27891


    H0 : Their is no significant difference in mean purchase value of the male and female
    
    H1 : Their is a significant difference in mean purchase value of the male and female
    
    Since p-value of 0.27891 is greater than 0.05(alpha value)
    
    Conclusion: we accept the null Hypothesis(H0) that their is no significant difference in mean purchase value of 
    the male and female
    
    Meaning: the male customers do not purchase more than the female customers

### Q4. Do customers acquired via email spend more than those from social channels?

In [9]:
email = customers_transactions[customers_transactions["acquisition_channel"] == 'email']["purchase_value"]
social = customers_transactions[customers_transactions["acquisition_channel"] == 'social']["purchase_value"]

t_stat, p_value = ttest_ind(email, social)

print(f"t-statistic: {t_stat:.3f}")
print(f"p-value: {p_value:.5f}")


t-statistic: -0.698
p-value: 0.48517


    H0 : Their is no significant difference in mean purchase value of the email customers and social channels customers
    
    H1 : Their is a significant difference in mean purchase value of the email customers and social channels customers
    
    Since p-value of 0.48517 is greater than 0.05(alpha value)
    
    Conclusion: we accept the null Hypothesis(H0) that their is no significant difference in mean purchase value of 
    the email customers and social channel customers 
    
    Meaning: customers acquired via email do not spend more than those acquired via social_channel

### Q5. Is device type associated with conversion?

In [10]:
device_type_by_converted = pd.crosstab(web_sessions["device_type"], web_sessions["converted"])
device_type_by_converted

converted,0,1
device_type,,
desktop,20020,3068
mobile,20213,3042
tablet,20244,3013


In [11]:
chi2, p, dof, expected = chi2_contingency(device_type_by_converted)

print(f"Chi-square statistic: {chi2:.3f}")
print(f"P-value: {p:.5f}")


Chi-square statistic: 1.150
P-value: 0.56265


    H0 : Their is no significant relationship(association) between device type and conversion
    
    H1 : Their is a significant relationship(association) between device type and conversion
    
    Since p-value of 0.56265 is greater than 0.05(alpha value)
    
    Conclusion: we accept the null Hypothesis(H0) that their is no significant relationship(association) between 
    device type and conversion

    Meaning: change in device type do not bring about any significant change in conversion

### Q6. Is acquisition channel associated with conversion?

In [12]:
customers_web_sessions = web_sessions.merge(customers, on='customer_id')
customers_web_sessions.head()

,session_id,customer_id,pages_viewed,time_on_site,device_type,traffic_source,session_date,converted,age,gender,country,acquisition_channel,signup_date
0,1,9716,8,271.238207,desktop,organic,2022-06-18,0,57.0,Other,Canada,organic,2020-11-04
1,2,4617,5,257.386842,mobile,social,2020-07-03,0,27.0,Female,Australia,social,2023-07-02
2,3,1506,4,289.679888,tablet,paid,2020-12-12,0,24.0,Female,Germany,organic,2019-10-11
3,4,3869,5,146.543376,mobile,direct,2023-09-12,0,69.0,Female,Germany,Email,2021-12-19
4,5,6245,3,339.228209,mobile,email,2019-04-02,0,NaN,Other,Canada,paid_search,2019-05-20


In [13]:
acquisition_channel_by_converted = pd.crosstab(customers_web_sessions["acquisition_channel"], customers_web_sessions["converted"])
acquisition_channel_by_converted

converted,0,1
acquisition_channel,,
Email,1708,277
email,11817,1771
organic,11079,1673
paid_Search,1206,160
paid_search,12003,1754
referral,11563,1746
social,11545,1798


In [14]:
chi2, p, dof, expected = chi2_contingency(acquisition_channel_by_converted)

print(f"Chi-square statistic: {chi2:.3f}")
print(f"P-value: {p:.5f}")


Chi-square statistic: 6.775
P-value: 0.34213


    H0 : Their is no significant relationship(association) between acquisition_channel and conversion
    
    H1 : Their is a significant relationship(association) between acquisition channel and conversion
    
    Since p-value of 0.34213 is greater than 0.05(alpha value)
    
    Conclusion: we accept the null Hypothesis(H0) that their is no significant relationship(association) 
    between acquisition channel and conversion
    
    Meaning: changes in acquisition channel does not bring about any significant change in conversion

### Q7. Is product category associated with discount usage?

In [15]:
product_category_by_discount = pd.crosstab(transactions["product_category"], transactions["discount_used"])
product_category_by_discount 

discount_used,0,1
product_category,,
beauty,5305,2281
electronics,5212,2212
fashion,5326,2255
home,5210,2278
sports,5225,2279
toys,5196,2221


In [16]:
chi2, p, dof, expected = chi2_contingency(product_category_by_discount)

print(f"Chi-square statistic: {chi2:.3f}")
print(f"P-value: {p:.5f}")


Chi-square statistic: 1.462
P-value: 0.91739


    H0 : Their is no relationship(association) between product category and discount usage
    
    H1 : Their is a relationship(association) between product category and discount usage
    
    Since p-value of 0.91739 is greater than  0.05(alpha value)
    
    Conclusion: we accept the null Hypothesis(H0) that their is no significant relationship(association) between 
    product category and discount usage
    
    Meaning: changes in product category does not bring about any significant change in discount usage

### Q8. Is time on site correlated with purchase value?

In [17]:
transactions_web_sessions = web_sessions.merge(transactions, on='customer_id')
transactions_web_sessions.head()

,session_id,customer_id,pages_viewed,time_on_site,device_type,traffic_source,session_date,converted,transaction_id,product_category,purchase_value,discount_used,purchase_date
0,1,9716,8,271.238207,desktop,organic,2022-06-18,0,5573,fashion,107.82,0,2022-09-04
1,1,9716,8,271.238207,desktop,organic,2022-06-18,0,5674,fashion,42.80,0,2023-08-30
2,1,9716,8,271.238207,desktop,organic,2022-06-18,0,6169,toys,111.01,0,2019-02-18
3,1,9716,8,271.238207,desktop,organic,2022-06-18,0,6933,sports,181.87,1,2021-02-07
4,1,9716,8,271.238207,desktop,organic,2022-06-18,0,7607,home,45.22,0,2023-09-29


In [18]:
corr = transactions_web_sessions[["purchase_value","time_on_site","pages_viewed"]].corr()
corr

,purchase_value,time_on_site,pages_viewed
purchase_value,1.000000,-0.001672,0.000092
time_on_site,-0.001672,1.000000,0.004397
pages_viewed,0.000092,0.004397,1.000000


In [19]:
corr, p_value = pearsonr(transactions_web_sessions["purchase_value"], transactions_web_sessions["time_on_site"])

print(f"Correlation: {corr:.3f}")
print(f"p-value: {p_value:.5f}")


Correlation: -0.002
p-value: 0.34794


    H0 : Their is no significant relationship between time on site and purchase value
    
    H1 : Their is a significant relationship between time on site and purchase value
    
    Since p-value of 0.34794 is greater than  0.05(alpha value)
    
    Conclusion: we accept the null Hypothesis(H0) that their is no significant relationship between time on site 
    and purchase value
    
    With R-value = -0.002, R-squared = 0.000004, Percentage value = 0.0004%
    
    Meaning: 0.0004%  of change in purchase value resulted from change in time on site  

### Q9. Is number of pages viewed correlated with purchase value?

In [20]:
corr, p_value = pearsonr(transactions_web_sessions["purchase_value"], transactions_web_sessions["pages_viewed"])

print(f"Correlation: {corr:.3f}")
print(f"p-value: {p_value:.5f}")


Correlation: 0.000
p-value: 0.95879


    H0 : Their is no relationship between number of pages viewed and purchase value
    
    H1 : Their is a relationship between number of pages viewd and purchase value
    
    Since p-value of 0.95879 is greater than  0.05(alpha value)
    
    Conclusion  1: we accept the null Hypothesis(H0) that their is no significant relationship between number of 
    pages viewed and purchase value
    
    With R-value = 0.000, R-squared = 0.00000, Percentage value = 0.000%
    
    Meaning: 0.000%  of change in purchase value resulted from the change in the number of pages viewed

### Q10. Is time on site correlated with pages viewed?

In [21]:
corr, p_value = pearsonr(transactions_web_sessions["time_on_site"], transactions_web_sessions["pages_viewed"])

print(f"Correlation: {corr:.3f}")
print(f"p-value: {p_value:.5f}")


Correlation: 0.004
p-value: 0.01356


    H0 : Their is no significant relationship between time on site and number of pages viewed
    
    H1 : Their is a significant relationship between time on site and number of pages viewed
    
    Since p-value of 0.01356 which is less than  0.05(alpha value)
    
    Conclusion  1: we accept the alternative Hypothesis(H1) that their is a significant relationship between time on 
    site and number of pages viewed
    
    With R-value = 0.004, R-squared = 0.000016, Percentage value = 0.0016%
    
    Meaning: 0.0016%  of the change in number of pages viewed resulted from change in the number of time on site

### Challenges

### C1. Explore whether high-spending customers behave differently in terms of session time.

### Define “high spending” yourself: 
        
        high spending          : purchase value greater than or equal to 1000
        low spending           : purchase value less than 500

In [22]:
high_spending = transactions_web_sessions[transactions_web_sessions["purchase_value"] >= 1000]["time_on_site"]
low_spending = transactions_web_sessions[transactions_web_sessions["purchase_value"] < 500]["time_on_site"]

t_stat, p_value = ttest_ind(high_spending, low_spending)

print(f"t-statistic: {t_stat:.3f}")
print(f"p-value: {p_value:.5f}")

t-statistic: -0.007
p-value: 0.99474


    H0 : Their is no significant difference in mean time on site of high spenders and low spendings
    
    H1 : Their is significant difference in mean time on site of high spenders and low spendings
    
    Since p-value of 0.99474 is greater than 0.05(alpha value)
    
    Conclusion: we accept the null Hypothesis(H0) that their is no significant difference in mean time on site of 
    high spenders and low spenders
    
    Meaning that: customers who spend high does not use more time on site than customer who spend low

### C2. Find a relationship (difference or correlation) that you believe is meaningful for business strategy.

### Explain why.

In [23]:
discount = transactions[transactions["discount_used"] == 1]["purchase_value"]
no_discount = transactions[transactions["discount_used"] == 0]["purchase_value"]

t_stat, p_value = ttest_ind(discount, no_discount)

print(f"t-statistic: {t_stat:.3f}")
print(f"p-value: {p_value:.5f}")

t-statistic: -27.160
p-value: 0.00000


    H0 : Their is no significant difference in mean purchase value of the discounted and non-discounted

    H1 : Their is a significant difference in mean purchase value of the discounted and non-discounted

    Since p-value of 0.00000 is less than  0.05(alpha value)

    Conclusion: we accept the alternative Hypothesis(H1) that their is a significant difference in mean purchase value 
    of the discounted and non-discounted.

    Meaning: the discounted average purchase value differs to the average purchase value of non-discounted.

### Explain why.
    Giving a discount means asking customers to pay a lower price for your product. 

    Provided product's qaulity and standard is maintain, the law of demand says; the higher the price the lower the 
    demand and vice versa.

    With proper marketing(advertisement) to create awareness, discount will raise demand thus increasing  total 
    purchase value(total revenue).

### c3. Identify a statistically significant result that may not be practically meaningful.

### Explain why statistical significance ≠ business importance.

In [24]:
traffic_source_by_converted = pd.crosstab(transactions_web_sessions["traffic_source"], transactions_web_sessions["converted"])
traffic_source_by_converted

converted,0,1
traffic_source,,
direct,55395,8193
email,54935,8461
organic,54357,8350
paid,55286,7965
social,54290,8015


In [25]:
chi2, p, dof, expected = chi2_contingency(traffic_source_by_converted)

print(f"Chi-square statistic: {chi2:.3f}")
print(f"P-value: {p:.5f}")

Chi-square statistic: 23.298
P-value: 0.00011


    H0 : Their is no relationship(association) between traffic source and converted.

    H1 : Their is a relationship(association) between traffic source and converted.

    Since p-value of 0.00011 is less than  0.05(alpha value).

    Conclusion: we accept the alternative Hypothesis(H1) that their is significant relationship(association) between 
    traffic source and converted.

    Meaning: change in traffice source does bring about significant change in conversion.

### Explain why statistical significance ≠ business importance.
    
    1. Statitistical significance tells if difference exist, not whether that difference actually matters in the real world.
    
    2. The cost of implementing that change or difference may not be justified in real world.
      
    3. With large sample, a tiny and irrelevant difference can become statistically significant.
    
    4  Using accurate data, a tiny inconsequential effect can bring about low p-value.